tls/extensions/s2n_client_psk.c

```diff
@@ -299,9 +299,6 @@ int s2n_client_psk_recv(struct s2n_connection *conn, struct s2n_stuffer *extensi
         GUARD_AS_POSIX(s2n_client_psk_recv_binders(conn, extension));
     }
 
-    /* At this point, we have either chosen a PSK or fallen back to a full handshake.
-     * Wipe any PSKs not chosen. */
-    GUARD_AS_POSIX(s2n_psk_parameters_free_unused_psks(&conn->psk_params));
-
+    /* At this point, we have either chosen a PSK or fallen back to a full handshake. */
```

Comment:
It's not really true on line 292. At that point, we've chosen a PSK but we could fail because the binders aren't valid. Until the end of this method we don't know we've either successfully chosen a PSK or fallen back.